cryostatio/cryostat-agent
# fix(http): use Apache HTTP client

Fixes #54

Replaces the JDK HTTP Client with Apache HTTP Client. This is a more robust and easier to use library, and also fixes the above bug.

To test:
1. `mvn install` to get the JAR into local mvn repo
2. `cd quarkus-test ; ./mvnw package && podman build -t quay.io/andrewazores/quarkus-test:latest -f src/main/docker/Dockerfile.jvm . ; podman image prune -f` to build a `quarkus-test` using this updated agent JAR
3. `cd cryostat ; sh smoketest.sh` to run the smoketest setup. `quarkus-test-agent-{0,1,2}` should spin up, each using the new Agent JAR containing this change. `0` should fail to register itself due to a missing environment variable, but the host application should run as usual. `1` should register itself successfully and then periodically push harvested JFR files. `2` should register itself and do nothing further.
4. `^C` on smoketest.sh to tear down. All three `quarkus-test-agent` instances should cleanly exit along with everything else.
